klippy/mcu.py

```diff
@@ -180,15 +180,17 @@ def _build_config(self):
                                , self._oid)
     def home_prepare(self):
         pass
-    def home_start(self, print_time, sample_time, sample_count, rest_time):
+    def home_start(self, print_time, sample_time, sample_count, rest_time,
+                   triggered=True):
         clock = self._mcu.print_time_to_clock(print_time)
         rest_ticks = int(rest_time * self._mcu.get_adjusted_freq())
         self._homing = True
         self._min_query_time = self._mcu.monotonic()
         self._next_query_time = self._min_query_time + self.RETRY_QUERY
         self._home_cmd.send(
             [self._oid, clock, self._mcu.seconds_to_clock(sample_time),
-             sample_count, rest_ticks, 1 ^ self._invert], reqclock=clock)
+             sample_count, rest_ticks, triggered ^ self._invert],
+            reqclock=clock)
```

Comment:
The purpose of multiple sampling is to filter out line noise on digital io pins.  Specifically, it's common for endstop/bltouch wiring to be run next to stepper wires - changes in current on the stepper wire can cause noise on the signaling wires that result in sporadic 1-2us high/low reports on the digital io pins.

An incorrect triggered reading on the bltouch sensor pin would result in the head stopping well before the nozzle is near the bed, and it would ruin the subsequent print.  The filtering prevents this and is a simple addition.

Reply:
I'm quite skeptical, since Marlin has been fine with reading the pin just once. But I suppose that you are right that there could be some people with very badly protected wires having problems.